src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Requests/DataSetGetQueryLocationsValidatorTests.cs

```diff
@@ -9,147 +9,154 @@ public abstract class DataSetGetQueryLocationsValidatorTests
 
-    public static IEnumerable<object[]> ValidLocationStringsMultiple()
-    {
-        return
-        [
-            ["NAT|id|12345", "NAT|code|E92000001"],
-            ["REG|id|12345", "REG|code|E12000003"],
-            ["LA|id|12345", "LA|code|E08000019", "LA|oldCode|373"],
-            ["SCH|id|12345", "SCH|urn|107029", "SCH|laEstab|3732060"],
-            ["PROV|id|12345", "PROV|ukprn|10066874"],
-        ];
-    }
+    public static readonly TheoryData<DataSetQueryLocation[]> ValidLocationQueriesMultiple =
+        DataSetQueryCriteriaLocationsValidatorTests.ValidLocationsMultiple;
 
-    public static IEnumerable<object[]> InvalidLocationStringsSingle()
+    public static readonly TheoryData<string> InvalidLocationFormatsSingle = new()
     {
-        return
-        [
-            [""],
-            ["Invalid"],
-            ["NA|id|12345"],
-            ["NAT|urn|12345"],
-            ["NAT|id|99999999999999999999999999"],
-        ];
-    }
-
-    public static IEnumerable<object[]> InvalidLocationStringsMultiple()
+        "",
+        "Invalid",
+        "NAT|",
+        "|",
+        "||"
+    };
+
+    public static readonly TheoryData<string[]> InvalidLocationFormatsMultiple = new()
     {
-        return
-        [
-            [],
-            ["", ""],
-            ["Invalid", "NAT|id", "NAT|code|"],
-            ["NA|id|12345", "la|code|12345"],
-            ["NAT|urn|12345", "REG|ukprn|12345", "RSC|code|12345"],
-            ["NAT|id|99999999999999999999999999", "NAT|code|99999999999999999999999999"],
-        ];
-    }
+        Array.Empty<string>(),
+        new [] { "", "" },
+        new [] { "Invalid", "NAT|", "|", "||" },
+    };
+
+    public static readonly TheoryData<DataSetQueryLocation> InvalidLocationQueriesSingle =
+        DataSetQueryCriteriaLocationsValidatorTests.InvalidLocationsSingle;
 
+    public static readonly TheoryData<DataSetQueryLocation[]> InvalidLocationQueriesMultiple =
+        DataSetQueryCriteriaLocationsValidatorTests.InvalidLocationsMultiple;
+    
     public class EqTests : DataSetGetQueryLocationsValidatorTests
     {
         [Theory]
-        [MemberData(nameof(ValidLocationStringsSingle))]
-        public void Success(string location)
+        [MemberData(nameof(ValidLocationQueriesSingle))]
+        public void Success(DataSetQueryLocation location)
         {
-            var query = new DataSetGetQueryLocations { Eq = location };
+            var query = new DataSetGetQueryLocations { Eq = location.ToLocationString() };
 
             _validator.TestValidate(query).ShouldNotHaveAnyValidationErrors();
         }
 
         [Theory]
-        [MemberData(nameof(InvalidLocationStringsSingle))]
+        [MemberData(nameof(InvalidLocationFormatsSingle))]
         public void Failure_InvalidString(string location)
         {
             var query = new DataSetGetQueryLocations { Eq = location };
 
             _validator.TestValidate(query)
```

Comment:
we could check for the specific error details here couldn't we? For instance, here I think it would be `ValidationMessages.LocationFormat.Code` etc.

Reply:
Nope, don't think so as we're testing a mixture of cases in the theory data so it's not simple to test every type of error that can occur.

Additionally, we already test the exact errors in great detail in `LocationStringValidatorTests` so this would be needlessly redundant.

Think it'd be better to keep these tests fairly high level just to prove integration and leave it at that.